CMake/RealmCore.cmake

```diff
@@ -272,22 +281,44 @@ endfunction()
 
 macro(build_realm_sync)
     set(cmake_files ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+    if(REALM_PLATFORM STREQUAL "Android")
+        set(build_cmd sh build.sh build-android)
+    else()
+        set(build_cmd make -C src/realm librealm-sync.a librealm-sync-dbg.a librealm-server.a librealm-server-dbg.a ${MAKE_FLAGS})
+    endif()
+
     ExternalProject_Add(realm-sync-lib
         DEPENDS realm-core
         PREFIX ${cmake_files}/realm-sync
         BUILD_IN_SOURCE 1
-        BUILD_COMMAND make -C src/realm librealm-sync.a librealm-sync-dbg.a librealm-server.a librealm-server-dbg.a ${MAKE_FLAGS}
+        BUILD_COMMAND ${build_cmd}
         CONFIGURE_COMMAND ""
         INSTALL_COMMAND ""
         ${USES_TERMINAL_BUILD}
         ${ARGV}
         )
 
+    if(APPLE)
+        set(platform "")
+    elseif(REALM_PLATFORM STREQUAL "Android")
+        if(ANDROID_ABI STREQUAL "armeabi-v7a")
+            set(platform "-android-arm-v7a")
+        else()
+            set(platform "-android-${ANDROID_ABI}")
+        endif()
+    endif()
+
+    
     ExternalProject_Get_Property(realm-sync-lib SOURCE_DIR)
     set(sync_directory ${SOURCE_DIR})
 
-    set(sync_library_debug ${sync_directory}/src/realm/librealm-sync-dbg.a)
-    set(sync_library_release ${sync_directory}/src/realm/librealm-sync.a)
+    if(REALM_PLATFORM STREQUAL "Android")
```

Comment:
The path to the built libraries could also be hoisted into the Android-vs-other if block above, so we don't have to duplicate the list of libraries for both sync and the sync server.